graphql/batch_executor.go

```diff
@@ -335,6 +355,16 @@ func resolveObjectBatch(ctx context.Context, sources []interface{}, typ *Object,
 		}
 
 		field := typ.Fields[selection.Name]
+		if field.Batch && selection.UseBatch {
+			workUnits = append(workUnits, &WorkUnit{
+				ctx:          ctx,
```

Comment:
WorkUnit here looks duplicated a bunch of times which distracts me from the actual logic. How about you create the WorkUnit once, and then later decide where/how to append it?

For the field.Expensive case, you could have a helper that splits a batch work unit into an array of work units.